components/entities/api/src/main/java/org/phenotips/entities/PrimaryEntity.java

```diff
@@ -53,7 +55,14 @@
      *
      * @return a valid document reference
      */
-    DocumentReference getDocument();
+    DocumentReference getDocumentReference();
+
+    /**
+     * Returns the document where the entity is stored.
+     *
+     * @return a valid document reference
+     */
+    XWikiDocument getDocument();
```

Comment:
+1 for `SecurePatientRepository` returning a special Patient which returns `null` for `getXWikiDocument()`.

...but, for scripts, do we expect any clients to have velocity code? if yes then I guess we are back to square 0 with keeping the old API, if no we should change those as well, and possibly optimize scripts the same way we have to optimize old uses for `getDocument()` in Java